src/inject/inject.js

```diff
@@ -9,16 +9,35 @@ const NAME = "test";
 const addElement = (e) => {
   acknowledge(e);
 
-  const element = document.createElement('span');
-  element.textContent = '⚠';
-  element.title = e.detail.type;
-  
+  const target = e.target.querySelector(`[data-name="${NAME}"]`);
+
+  // Events are received that are intended for all consumers. 
+  // This makes sure the extension only handles intended for it.
```

Comment:
This isn't quite accurate. If the consumer registered a name, then the JSAPI should render a named container for it. This sanity check only seems useful for either guarding against forgetting the `<meta name="jsapi.consumer" content="name" />` content attribute, or something breaking in the JSAPI.

Reply:
You are right, in that case I should report a jsAPI defect as it is happening.

Reply:
A-hah! Yes, race condition makes sense.